src/rustup/toolchain.rs

```diff
@@ -381,6 +382,7 @@ impl<'a> Toolchain<'a> {
                     component: component.clone(),
                     required: true,
                     installed: installed,
+                    available: targ_pkg.available,
```

Comment:
This is almost there, but not quite. The `TargettedPackage` represented by `targ_pkg` is the "rust" package, but you need the `TargettedPackage` for each of the components of "rust" (the "rust" package is made up of other components like "rust-std", each with their own packages).

So in both these loops over components you're going to need to take the `pkg` and `target`  fields of the component and then look up the targetted package in the manifest *for that component*. This is done the same way as the lookups for the previous `rust_pkg` and `targ_pkg` variables.

Thanks!